src/Evolution/DiscontinuousGalerkin/LiftFromBoundary.hpp

```diff
@@ -0,0 +1,179 @@
+    const DataVector& boundary_lifting_term,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const Scalar<DataVector>& face_det_jacobian) noexcept;
+
+template <size_t Dim>
+void lift_boundary_terms_gauss_points_impl(
+    gsl::not_null<double*> volume_dt_vars, size_t num_independent_components,
+    const Mesh<Dim>& volume_mesh, size_t dimension,
+    const gsl::span<const double>& volume_det_inv_jacobian,
+    size_t num_boundary_pts,
+    const gsl::span<const double>& upper_boundary_corrections,
+    const DataVector& upper_boundary_lifting_term,
+    const Scalar<DataVector>& upper_magnitude_of_face_normal,
+    const Scalar<DataVector>& upper_face_det_jacobian,
+    const gsl::span<const double>& lower_boundary_corrections,
+    const DataVector& lower_boundary_lifting_term,
+    const Scalar<DataVector>& lower_magnitude_of_face_normal,
+    const Scalar<DataVector>& lower_face_det_jacobian) noexcept;
+}  // namespace detail
+
+/*!
+ * \brief Lift the boundary corrections to the volume time derivatives in the
+ * specified direction.
+ *
+ * The `boundary_corrections` are lifted to the `dt_vars` from the face in the
+ * direction `direction`. The magnitude of the normal vector in the \f$\xi\f$
+ * direction, for example, is given by
+ *
+ * \f{align*}{
+ *  \sqrt{\frac{\partial \xi}{\partial x^i} \gamma^{ij}
+ *  \frac{\partial \xi}{\partial x^j}}
+ * \f}
+ *
+ * The general lifting term (for the upper \f$\xi\f$-direction) is:
+ *
+ * \f{align*}{
+ * \partial_t u_{\breve{\imath}\breve{\jmath}\breve{k}}=\cdots
+ * -\frac{\ell_{\breve{\imath}}\left(\xi=1\right)}
+ *   {w_{\breve{\imath}}J_{\breve{\imath}\breve{\jmath}\breve{k}}}
+ *   \left[J\sqrt{
+ *   \frac{\partial\xi}{\partial x^i} \gamma^{ij}
+ *   \frac{\partial\xi}{\partial x^j}}
+ *   \left(G + D\right)
+ *   \right]_{\breve{\jmath}\breve{k}}\left(\xi=1\right),
+ * \f}
+ *
+ * where \f$\breve{\imath}\f$, \f$\breve{\jmath}\f$, and \f$\breve{k}\f$ are
+ * indices in the logical \f$\xi\f$, \f$\eta\f$, and \f$\zeta\f$ directions.
+ * The \f$G+D\f$ term corresponds to the `boundary_corrections`  function
+ * argument, and the function Spectral::boundary_lifting_term() is used to
+ * compute and cache the terms from the lifting term
+ * \f$\ell_{\breve{\imath}}(\xi=1)/w_{\breve{\imath}}\f$.
+ *
+ * \note that normal vectors are pointing out of the element and therefore the
+ * boundary term has an overall negative sign.
+ */
+template <size_t Dim, typename DtTagsList, typename BoundaryCorrectionTagsList>
+void lift_boundary_terms_gauss_points(
+    const gsl::not_null<Variables<DtTagsList>*> dt_vars,
+    const Scalar<DataVector>& volume_det_inv_jacobian,
+    const Mesh<Dim>& volume_mesh, const Direction<Dim>& direction,
+    const Variables<BoundaryCorrectionTagsList>& boundary_corrections,
+    const Scalar<DataVector>& magnitude_of_face_normal,
+    const Scalar<DataVector>& face_det_jacobian) noexcept {
+  const Mesh<Dim - 1> boundary_mesh =
```

Comment:
Since this is `lift_boundary_terms_gauss_points`, I would suggest the first thing in the function should be some assertions that `volume_mesh` is indeed using Gauss points.

Same for "two at a time" function below.